crates/bevy_ecs/src/query/state.rs

```diff
@@ -840,12 +846,12 @@ impl<Q: WorldQuery, F: WorldQuery> QueryState<Q, F> {
     >(
         &self,
         world: &'w World,
-        task_pool: &TaskPool,
         batch_size: usize,
         func: FN,
         last_change_tick: u32,
         change_tick: u32,
     ) {
+        let task_pool = world.resource::<ComputeTaskPool>().clone();
```

Comment:
This could be unsafe, as other systems might require `ResMut<ComputeTaskPool>`. The task pool should be fetched at `QueryState` creation and stored as `Option<TaskPool>`.

Reply:
This does create an initialization order issue. `ComputeTaskPool` must be initialized before any `Query(State)` or any later `par_for_each` call will panic, even if it was initialized later.

Made the change for safety.